projects/distributed-tracing/src/pages/trace-detail/trace-detail.service.ts

```diff
@@ -109,15 +125,26 @@ export class TraceDetailService implements OnDestroy {
 
     return '';
   }
+
+  protected buildGraphqlTimeRange(timeRange: TimeRange, startTime?: unknown): GraphQlTimeRange {
+    const startTimeAsDate = this.dateCoercer.coerce(startTime);
+
+    return startTimeAsDate !== undefined
+      ? new GraphQlTimeRange(startTimeAsDate.getTime() - 1, startTimeAsDate.getTime() + 1)
+      : new GraphQlTimeRange(timeRange.startTime, timeRange.endTime);
+  }
 }
 
 interface TraceDetailRouteIdParams {
   traceId: string;
   spanId?: string;
+  startTime?: unknown;
```

Comment:
why unknown? params are always strings right?

Reply:
It can be number, string or undefined. I kept it unknown since we are using dateCoercer anyway. But I can change it to just `string | number` if you like

Reply:
Even if it's going to a date coercer, being more specific on the types earlier in the chain I think is still a gppd thing . These values are always originating from `paramMap.get` though right? https://angular.io/api/router/ParamMap#get looks like it can only return a string, if anything.

Reply:
done